hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -186,12 +188,11 @@ public ReplicationSourceManager(ReplicationQueueStorage queueStorage,
     this.latestPaths = new HashMap<>();
     this.replicationForBulkLoadDataEnabled = conf.getBoolean(
       HConstants.REPLICATION_BULKLOAD_ENABLE_KEY, HConstants.REPLICATION_BULKLOAD_ENABLE_DEFAULT);
-    this.sleepForRetries = this.conf.getLong("replication.source.sync.sleepforretries", 1000);
-    this.maxRetriesMultiplier =
-      this.conf.getInt("replication.source.sync.maxretriesmultiplier", 60);
     this.totalBufferLimit = conf.getLong(HConstants.REPLICATION_SOURCE_TOTAL_BUFFER_KEY,
         HConstants.REPLICATION_SOURCE_TOTAL_BUFFER_DFAULT);
     this.globalMetrics = globalMetrics;
+    this.replicationOffload = conf.getBoolean(HConstants.REPLICATION_OFFLOAD_ENABLE_KEY,
```

Comment:
I should go back to the design but we can't just have the offload be a ReplicationSource implementation? Because you need to span all ReplicationSources?

Reply:
In the design, the RegionServer not need to start all ReplicationSources. It only need to store its WAL path to the ReplicationQueueStorage. And the ReplicationServer will fetch the WAL paths and start all ReplicationSources.